.github/workflows/self-test.yml

```
@@ -0,0 +1,59 @@
+name: Self test action
+
+on:
+  push:
+    branches: main
+    paths-ignore: "docs/**"
+  pull_request:
+    branches: main
+    paths-ignore: "docs/**"
+
+jobs:
+  test:
+    strategy:
+      matrix:
+        os: [ ubuntu-latest, macos-latest, windows-latest ]
+        clang-version: ['9','10', '11', '12', '13', '14', '15', '16', '17', '18']
+      fail-fast: false
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Cache the build artifacts
+        id: cache-build
+        uses: actions/cache@v4
+        with:
+          path: build
+          key: ${{ runner.os }}-${{ hashFiles('docs/examples/demo/CMakeLists.txt', 'docs/examples/demo/demo.cpp', 'docs/examples/demo/demo.hpp') }}
```

```suggestion
          key: ${{ runner.os }}-${{ hashFiles('docs/examples/demo/**') }}
```
Could be easier to glob all files in the demo folder. This way any changes to that folder will cause the cache to be refreshed.